### PR TITLE
Suport sliced with Enumerator::ArithmeticSequence on Array

### PR DESCRIPTION
**Pull request by edipofederle**
WIP

### CODE REVIEW

**Comment by headius** [comment]
This feature is the only thing not passing on master's 3.0 specs. Let us know if you want some help getting it finished!

**Comment by edipofederle** [comment]
> This feature is the only thing not passing on master's 3.0 specs. Let us know if you want some help getting it finished!

Hi @headius,

I'm planing back to work on it today or tomorrow, I let you know if I need something here.

File: core/src/main/java/org/jruby/RubyArray.java [hunk 1]
```diff
@@ -1283,6 +1283,116 @@ public IRubyObject subseq(long beg, long len) {
...
+        if (end < 0) {  end += len; }
+
+        if (!aseqExcl) {  end++; }
+
+        len = end - beg;
+
+        if (len < 0) {  len = 0;  }
+
+        long alen = realLength;
+        if (beg > alen)  return null;
```
Comment (enebo) [inline core/src/main/java/org/jruby/RubyArray.java:1332]: Some null returns here.  I think these should be nil?

**Review by enebo — CHANGES_REQUESTED** [review]
Also I mentioned using subseq in chat.

File: core/src/main/java/org/jruby/RubyArray.java (hunk 1 above)
Comment (edipofederle) [reply core/src/main/java/org/jruby/RubyArray.java]: done

**Comment by enebo** [comment]
Thanks @edipofederle!  Keep these PRs coming :)

**Pull request merged**